### PR TITLE
New lint and analysis cleanup with Dart 2.17 and lints 2.0.0

### PR DESCRIPTION
**Pull request by parlough**

### CODE REVIEW

File: lib/elements/elements.dart [hunk 1]
```diff
@@ -415,12 +415,12 @@ class TabElement extends DElement {
...
+  TabElement(super.element, {this.name, this.onSelect});
 
   void handleSelected() {
-    if (onSelect != null) onSelect!();
+    onSelect?.call();
   }
 
   @override
-  String toString() => name!;
+  String toString() => '$name';
```
Comment (domesticmouse) [inline lib/elements/elements.dart:425]: Does this convert an exception to `"null"`?

File: web/example/picker.dart [hunk 2]
```diff
@@ -79,13 +79,10 @@ class DartPadPicker {
     _iFrameElement = IFrameElement()
       ..src = iFrameSrc(theme: 'dark', mode: 'dart');
     iFrameHost.children.add(_iFrameElement);
-    window.addEventListener('message', (Event _e) {
-      final e = _e as MessageEvent;
+    window.addEventListener('message', (Event event) {
+      final messageEvent = event as MessageEvent;
```
Comment (domesticmouse) [inline web/example/picker.dart:83]: I'd think about converting this to `if (event is MessageEvent) { ... }`

File: lib/elements/elements.dart (hunk 1 above)
Comment (parlough) [reply lib/elements/elements.dart]: Yes it does. I switched to this to match the original functionality of the code, but after a second look, these fields don't need to be nullable anyway. I've adjusted the class in the latest commit :)

**Comment by domesticmouse** [comment]
nice, LGTM

**Pull request merged**